Mathlib/Data/List/Destutter.lean

```diff
@@ -173,4 +185,187 @@ theorem destutter_eq_nil : ∀ {l : List α}, destutter R l = [] ↔ l = []
   | _ :: l => ⟨fun h => absurd h <| l.destutter'_ne_nil R, fun h => nomatch h⟩
 #align list.destutter_eq_nil List.destutter_eq_nil
 
+/-- For a relation-preserving map, `destutter'` commutes with `map`. -/
+theorem destutter_map_iff (f : α → β) (h : ∀ a b, R a b ↔ R₂ (f a) (f b)) :
+    (l.destutter R).map f = (l.map f).destutter R₂ := by
+  cases h2 : l with
+  | nil => simp -- l = []
+  | cons a as =>
+    clear h2
+    induction as generalizing a with
+    | nil => simp -- l = a :: []
+    | cons a2 bs ih => -- l = a :: a2 :: bs
+      repeat rw [map_cons, destutter_cons_cons]
+      simp_rw [← h a a2]
+      by_cases hr : (R a a2) <;>
+        simp [hr, ← destutter_cons', ih]
+
+/-- For a injective function `f`, `destutter' (·≠·)` commutes with `map f`. -/
+theorem map_destutter_ne {f : α → β} (h : Injective f) [DecidableEq α] [DecidableEq β] :
+    (l.destutter (·≠·)).map f = (l.map f).destutter (·≠·) :=
+  destutter_map_iff l _ f (fun _ _ ↦ (Injective.ne_iff h).symm)
+
+/-- `destutter'` on a relation like ≠, whose negation is an equivalence, gives the same length if
+    the first elements are not related (¬Rα a b). --/
```

Comment:
Just FYI your `destutter'` won't turn into a link in the docs unless you "fully qualify" the name:
```suggestion
/-- `List.destutter'` on a relation like `≠`, whose negation is an equivalence, gives the same
length if the first elements are not related. -/
```
Also, docstrings end with `-/`, not `--/`. Be careful! Else you'll end up with docstrings with a trailing `-`.